### DISCUSSION TITLE
How do i grep for multiple strings

### DISCUSSION THREAD

**Discussion by Venomish-Snake** [BurntSushi/ripgrep#2124] [Q&A] [answered]
Hey all, Ive been struggling to find the answer on how to search for multiple strings or specified list of strings.

Lets say i want to grep for any occurances of 'cat' or 'dog'. 
How do i do this without searching in two instances- one for cat and one for dog?

Thanks guys!

**Comment by s-p-turner** [comment]
I think this should do it:
rg 'cat|dog'

**Reply by Venomish-Snake** [reply]
Thanks so much for the response! <3

**Comment by BurntSushi** [comment, marked as answer]
Yeah there are three ways: `rg 'cat|dog'`, `rg -e cat -e dog` and:

```
$ printf "cat\ndog\n" > patterns
$ rg -f patterns
```

**Reply by Venomish-Snake** [reply]
Awesome, Thanks so much! :)
